mono/metadata/boehm-gc.c

```diff
@@ -1943,6 +1943,43 @@ mono_gchandle_is_in_domain_internal (MonoGCHandle gchandle, MonoDomain *domain)
 	return result;
 }
 
+/**
+ * mono_gchandle_is_in_domain_internal_lock_free:
+ * \param gchandle a GCHandle's handle.
+ * \param domain An application domain.
+ * 
+ * Lock free version of \c mono_gchandle_is_in_domain_internal. To be used only if the caller
```

Comment:
"Lock free" may not be the best name  - lock free can mean that it is thread safe but just doesn't use lock.  It looks like there is already [some use ](https://github.com/Unity-Technologies/mono/blob/3d5cc34e80c498864a4d97dba9ed9a3fa30238e6/mono/utils/lock-free-queue.h)of that in mono.

`mono_gchandle_is_in_domain_internal_unsafe`?